aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/BaseHandlerStd.java

```diff
@@ -451,13 +447,24 @@ static Map<String, Parameter> computeModifiedDBParameters(
         return modifiedParameters;
     }
 
-    protected ProgressEvent<ResourceModel, CallbackContext> resetAllParameters(
+    protected ProgressEvent<ResourceModel, CallbackContext> resetParameters(
             final ProgressEvent<ResourceModel, CallbackContext> progress,
             final AmazonWebServicesClientProxy proxy,
-            final ProxyClient<RdsClient> proxyClient
+            final ProxyClient<RdsClient> proxyClient,
+            final Map<String, Parameter> currentParameters,
+            final Map<String, Object> desiredParams
     ) {
+        // reset only parameters that are missing from desired params. If parameter is in desired param, its value will be updated anyway.
+        final Map<String, Parameter> toBeReset = currentParameters.keySet().stream()
+                .filter(p -> !desiredParams.containsKey(p))
```

Comment:
There is no need to reset all parameters except for the desired ones. We can only reset parameters that are not included in the desired ones and have been changed by the user. which `Source` will be `user`. Resetting a parameter that has the `Source` as 'Engine default' is pointless.
Docs: https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_Parameter.html

Reply:
The code above has the same effect. We only reset parameters which are missing from desired ones, but were present in current parameters retrieved via filtered describe call.

Reply:
So, if I understood correctly, after this change, we will no longer undo any changes made by the customer to parameters through the UI console or any other client (out of band). This is because these parameters are not found in the previous or desired state.

Reply:
Yes, just like in DBParameterGroup